app/models/services/locators/doi_paper_locator.rb

```diff
@@ -6,30 +6,79 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
+    crossref = Crossref.new
+    if (result = crossref.get_doi_metadata(self.locator_id))
+      data = result['message']
+      paper = self.import_data_to_paper(paper,data,'crossref')
+    end
+
     pubmed = Pubmed.new
-    if uid = pubmed.find_uid_by_doi(self.locator_id)
+    if (uid = pubmed.find_uid_by_doi(self.locator_id))
```

Comment:
So this is the same issue.  I guess it ends up being a UI question.  Essentially these if statements are to keep the user from hitting an error page if they enter pubmed ID or DOI that is not in the APIs, either it's not covered or they typed it in wrong.  The error messages are handled in the references controller based on what is spit out.  

It seems to make more sense to have an alert pop up and have them sent back to the list show page instead of a broken page?